src/ocr.py

```diff
@@ -0,0 +1,290 @@
+    original_img: np.ndarray = None
+    processed_img: np.ndarray = None
+    def __getitem__(self, key):
+        return super().__getattribute__(key)
+
+class Ocr:
+    def __init__(self):
+        self.I_1 = re.compile(r"(?<=[%I0-9\-+])I|I(?=[%I0-9\-+])")
+        self.II_U = re.compile(r"(?<=[A-Z])II|II(?=[A-Z])|1?=[a-z]")
+        self.One_I = re.compile(r"(?<=[A-Z])1|1(?=[A-Z])|1?=[a-z]")
+        self.OneOne_U = re.compile(r"(?<=[A-Z])11|11(?=[A-Z])|1?=[a-z]")
+        with open('assets/tessdata/ocr_errors.csv') as file:
+            self.ocr_errors = dict(csv.reader(file, skipinitialspace = False, delimiter = ',', quoting = csv.QUOTE_NONE))
+
+    """
+    OCR input processing functions:
+    """
+    def crop_pad(self, image: np.ndarray = None):
+        # crop
+        image = image[4: image.shape[0]-4, 5: image.shape[1]-5]
+        # re-pad
+        image = np.pad(image, pad_width=[(4, 4),(4, 4),(0, 0)], mode='constant')
+        return image
+
+    """
+    OCR functions:
+    """
+
+    def img_to_bytes(self, image: np.ndarray, colorspace: str = 'BGR'):
+        # Sets an OpenCV-style image for recognition: https://github.com/sirfz/tesserocr/issues/198
+        bytes_per_pixel = image.shape[2] if len(image.shape) == 3 else 1
+        height, width   = image.shape[:2]
+        bytes_per_line  = bytes_per_pixel * width
+
+        if bytes_per_pixel != 1 and colorspace != 'RGB':
+            # non-RGB color image -> convert to RGB
+            image = cv2.cvtColor(image, getattr(cv2, f'COLOR_{colorspace}2RGB'))
+        elif bytes_per_pixel == 1 and image.dtype == bool:
+            # binary image -> convert to bitstream
+            image = np.packbits(image, axis=1)
+            bytes_per_line  = image.shape[1]
+            width = bytes_per_line * 8
+            bytes_per_pixel = 0
+        # else image already RGB or grayscale
+
+        return image.tobytes(), width, height, bytes_per_pixel, bytes_per_line
+
+    def image_to_text(self,
+        images: Union[np.ndarray, List[np.ndarray]],
+        model: str = "engd2r_inv_th",
+        psm: int = 3,
+        word_list: str = "all_strings.txt",
+        scale: float = 1.0,
+        crop_pad: bool = True,
+        erode: bool = True,
+        invert: bool = True,
+        threshold: int = 25,
+        digits_only: bool = False,
+        fix_regexps: bool = True,
+        check_known_errors: bool = True,
+        check_wordlist: bool = True,
+        word_match_threshold: float = 0.9
+    ) -> list[str]:
+        """
+        Uses Tesseract to read image(s)
+        :param images (required): image or list of images to read in OpenCV format.
+            Use a list of images rather than looping over single images where possible for best performance.
+        :param model: OCR language model basename to use (in assets/tessdata folder)
+        :param psm: Tesseract PSM to use. 7=single uniform text line, 6=single block of text, 3=auto without orientation.
```

Comment:
Ideally I wouldn't expose magic numbers outside and would rather create an Enum that describes semantics of each of those PSMs. e.g.
```python
class Segmentation(Enum):
     Auto = 0,
     AutoNoOcrOrOSD = 1,
     ...etc (not sure I correctly made examples but the point is for API to be self-descriptive)
```

Reply:
There already is this in Tesserocr for the OEM and PSM classes but either way you need a reference--either to look up what number to use with the magic numbers or to look up what keywords to put after PSM or OEM. I found it less cumbersome this way personally. As you say though it would make the code more readable.